### PR TITLE
Json POST data req - reaserch, publications & prev functions.

### PR DESCRIPTION
**Pull request by YanaPatyuk**
1. Update all post - json req function to more generic version. #78 
2. Created add/set function for recived json data for reaserch.json file via POST   req. #66 
3. Created add/set function for recived json data for academic-publications.json file via POST req. #80 
4. Change 'add new course' to the add/set json data file via post (now calles add/set teaching file whitch contains courses data among other info). #77 
5. Change the exception handeling for more informative.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -353,10 +459,42 @@ def update_user():
...
+        if full_file:
+            FileHandler.write_to_json(json_data=data, path=folder_path + target_file_path)
+        else:
+            FileHandler.append_to_json(data_obj_to_append=data, path=folder_path + target_file_path)
+        return jsonify({"status": 200})
+    except ValidationError as validation_error:
+        return jsonify({"Error": validation_error.message}), 400
+    except Exception as e:
+        print(e)
+        return jsonify({"Error": "Server internal error, check server"}), 400
```
Comment (teddy4445) [inline main.py:495]: return the info inside "e" in the json you are sending to the user

**Review by teddy4445 — APPROVED** [review]
Looks fine to me...

**Comment by YanaPatyuk** [comment]
Added 3 more json files updating for about.html #82

File: main.py (hunk 1 above)
Comment (YanaPatyuk) [reply main.py]: Fixed in the last version to return str(e)

**Review by teddy4445 — APPROVED** [review]
the changes are fitting the scope and work fine in a manual check of each one of them

**Pull request merged**